jvm/src/test/scala/cats/tests/TaskSpec.scala

```diff
@@ -0,0 +1,159 @@
+package cats
+package jvm
+package tests
+
+import cats.tests.CatsSuite
+import cats.data.{Streaming,Xor}
+
+class TaskTests extends CatsSuite {
+
+  test("Task should not run side-effects until .unsafePerformIO() is called") {
+    var run: Int = 0
+    val t: Task[Int] = Task.later { run += 1; run }
+    run should be (0)
+    t.unsafePerformIO()
+    run should be (1)
+  }
+
+  test("Task should only evaulate side-effects once for Task.later") {
+    var run: Int = 0
+    val t: Task[Int] = Task.later { run += 1; run }
+    t.unsafePerformIO()
+    run should be(1)
+
+    t.unsafePerformIO()
```

Comment:
These tests won't work on Javascript. For one because you're using the synchronous `unsafePerformIO`. The advantage of building your own `Scheduler` is that you can then have a [TestScheduler](https://github.com/monixio/monix/blob/master/monix-execution/shared/src/main/scala/monix/execution/schedulers/TestScheduler.scala) that can simulate time. So you can do something like ...

```scala
val s = TestScheduler()
// ...

s.tick() // execute immediate tasks
s.tick(1.minute) // simulate the passage of time
```